x/wasm/internal/keeper/keeper.go

```diff
@@ -693,6 +702,58 @@ func (k Keeper) dispatchMessages(ctx sdk.Context, contractAddr sdk.AccAddress, i
 	return nil
 }
 
+// PinCode pins the wasm contract in wasmvm cache
+func (k Keeper) PinCode(ctx sdk.Context, codeID uint64) error {
+	codeInfo := k.GetCodeInfo(ctx, codeID)
+	if codeInfo == nil {
+		return sdkerrors.Wrap(types.ErrNotFound, "code info")
+	}
+
+	if err := k.wasmer.Pin(codeInfo.CodeHash); err != nil {
+		return sdkerrors.Wrap(types.ErrPinContractFailed, err.Error())
+	}
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.GetPinnedCodeIndexPrefix(codeID), []byte{})
+	return nil
+}
+
+// UnpinCode removes the wasm contract from wasmvm cache
+func (k Keeper) UnpinCode(ctx sdk.Context, codeID uint64) error {
+	codeInfo := k.GetCodeInfo(ctx, codeID)
+	if codeInfo == nil {
+		return sdkerrors.Wrap(types.ErrNotFound, "code info")
+	}
+	if err := k.wasmer.Unpin(codeInfo.CodeHash); err != nil {
+		return sdkerrors.Wrap(types.ErrUnpinContractFailed, err.Error())
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.GetPinnedCodeIndexPrefix(codeID))
+	return nil
+}
+
+// IsPinnedCode returns true when codeID is pinned in wasmvm cache
+func (k Keeper) IsPinnedCode(ctx sdk.Context, codeID uint64) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.GetPinnedCodeIndexPrefix(codeID))
+}
+
+// InitializePinnedCodes updates wasmvm to pin to cache all contracts marked as pinned
+func (k Keeper) InitializePinnedCodes(ctx sdk.Context) error {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.PinnedCodeIndexPrefix)
+	iter := store.Iterator(nil, nil)
+	for ; iter.Valid(); iter.Next() {
+		codeInfo := k.GetCodeInfo(ctx, sdk.BigEndianToUint64(iter.Value()))
```

Comment:
This is the inverse of `types.GetPinnedCodeIndexPrefix()`, right?
Maybe you can declare a function `types.ParsePinnedCodeIndexPrefix()` or something, so have the encode/decode steps next to each other